### PR TITLE
Remove argument and optional from libraries

### PR DESCRIPTION
**Pull request by m-nash**
Final clean up for these two issues

- https://github.com/Azure/autorest.csharp/issues/4245
- https://github.com/Azure/autorest.csharp/issues/4250

# Contributing to the Azure SDK

Please see our [CONTRIBUTING.md](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md) if you are not familiar with contributing to this repository or have questions.

For specific information about pull request etiquette and best practices, see [this section](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#pull-request-etiquette-and-best-practices).

### CODE REVIEW

File: sdk/appconfiguration/Azure.Data.AppConfiguration/src/ConfigurationClient.cs [hunk 1]
```diff
@@ -74,8 +74,14 @@ public ConfigurationClient(Uri endpoint, TokenCredential credential)
         /// <param name="options">Options that allow configuration of requests sent to the configuration store.</param>
         public ConfigurationClient(Uri endpoint, TokenCredential credential, ConfigurationClientOptions options)
         {
-            Argument.AssertNotNull(endpoint, nameof(endpoint));
-            Argument.AssertNotNull(credential, nameof(credential));
+            if (endpoint == null)
```
Comment (jsquire) [inline sdk/appconfiguration/Azure.Data.AppConfiguration/src/ConfigurationClient.cs:77]: This is not generated code, why are changes for generator issues being made here?

File: sdk/appconfiguration/Azure.Data.AppConfiguration/src/Models/ConfigurationSettingsFilter.Serialization.cs [hunk 2]
```diff
@@ -13,7 +13,7 @@ void IUtf8JsonSerializable.Write(Utf8JsonWriter writer)
             writer.WriteStartObject();
             writer.WritePropertyName("key");
             writer.WriteStringValue(Key);
-            if (Optional.IsDefined(Label))
+            if (Label != null)
```
Comment (jsquire) [inline sdk/appconfiguration/Azure.Data.AppConfiguration/src/Models/ConfigurationSettingsFilter.Serialization.cs:16]: Again, not generated code.  Why are we making changes here?

**Review by jsquire — CHANGES_REQUESTED** [review]
Would you please provide context in the description of this PR why changes are being made and how the files being changed were identified?   The linked issues refer to removing generator dependencies on shared source files, but changes are also being made to non-generated code here.

File: sdk/appconfiguration/Azure.Data.AppConfiguration/src/ConfigurationClient.cs (hunk 1 above)
Comment (m-nash) [reply sdk/appconfiguration/Azure.Data.AppConfiguration/src/ConfigurationClient.cs:77]: As part of the openai work we were asking remove shared source from the libraries this affects both generated and non-generated.

For extension methods we have no choice as it won't compile if the generated code uses one extension and the non generated uses another

**Pull request closed without merging**